redis/types.go

```diff
@@ -0,0 +1,37 @@
+package redis
+
+import (
+	"time"
+
+	"github.com/go-redis/redis"
+)
+
+// Config for redis connection
+// TODO Set all default values
+type Config struct {
+	Host string `yaml:"host" env:"REDIS_HOST" env-description:"Redis host"`
+	Port string `yaml:"port" env:"REDIS_PORT" env-description:"Redis port"`
+	Pass string `env:"REDISPASS" env-description:"Redis user password"`
+}
+
+// repository represent the repository model
+type repository struct {
+	Client redis.Cmdable
+}
+
+// Repository represent the repositories
+type Repository interface {
+	Set(key string, value interface{}, exp time.Duration) error
+	Get(key string) (string, error)
+}
+
+// Set attaches the redis repository and set the data
+func (r *repository) Set(key string, value interface{}, exp time.Duration) error {
+	return r.Client.Set(key, value, exp).Err()
+}
+
+// Get attaches the redis repository and get the data
+func (r *repository) Get(key string) (string, error) {
```

Comment:
I would like to have the ability to work with Redis lists

Reply:
I agree that this is the limit the functionality of the Redis. I think we can expose the r.Client rather than encapsulate them in the repository struct. For the func Connect will return `redis.Cmdable` instead.

Reply:
https://redis.uptrace.dev/#executing-unsupported-commands
I still like your approach, so we can just add the method in interface to execute custom commands

Reply:
https://github.com/go-redis/redis/blob/aa456e55a7213876dfa5e856fb8467237d822771/commands.go#L75

Redis.Cmdable interface has a great functionallity.
@calj what you can advise us? Return redis interface from `Connect` or our interface?